src/RenderManager.cpp

```diff
@@ -46,17 +46,33 @@ void RenderManager::ScaleTileset(FIBITMAP* fiTilesetBmp)
 	unsigned tilesetScaledHeight = FreeImage_GetHeight(fiTilesetBmp) / nonScaledTileLength * scaleFactor;
 
 	tilesetBmps.push_back(FreeImage_Rescale(fiTilesetBmp, tilesetScaledWidth, tilesetScaledHeight));
-
 	FreeImage_Unload(fiTilesetBmp);
 }
 
-void RenderManager::AddTilesetRawBits(BYTE* bits, int width, int height, int pitch, unsigned bpp,
-	unsigned red_mask, unsigned green_mask, unsigned blue_mask)
+void RenderManager::AddTileset(BYTE* tilesetMemoryPointer, std::size_t tilesetSize)
```

Comment:
I like this idea. However, I get the following error:

Error (active)	E0167	argument of type "const BYTE *" is incompatible with parameter of type "BYTE *"	OP2MapImager	OP2MapImager\src\RenderManager.cpp 39

I don't think that FreeImage_OpenMemory can take a const BYTE* unfortunately.

Reply:
I was thinking more in terms of the parameters to AddTileset, with the unpacking done inside the method, rather than at the caller.

The concern I'm having now, is my suggestion is less general. The memory might come from something other than a `vector`. Unpacked parameters allows for other data sources/memory buffer types.